lib/ojs.ml

```diff
@@ -13,7 +13,12 @@
 *)
 
 
-type t
+type t = Jsoo_runtime.Js.t
+
+external pure_js_expr: string -> t = "caml_pure_js_expr"
```

Comment:
Like @dbuenzli (See https://github.com/dbuenzli/brr/issues/45), I am not sure if exposing these functions as "external" in the interface would impact the performance or not.

Reply:
I would say it's better use externals because it allows jsoo to perform more optimisations.

Note that jsoo_runtime doesn't have an mli / it export external in its API.
I think it doesn't change any optimisation if you use the external defined locally or in another file/module

Reply:
My understanding is as follows:

1) when using the bytecode compiler `ocamlc`, if `ojs` does _not_ expose the `external` declaration, clients of this module will indeed pay a (small) penalty, roughly one function call + one memory indirection.
2) however, JSOO is clever enough to inline the external declaration across modules, which removes the penalty when targeting Javascript.

Conclusion: you can reference the externals declared in `jsoo_runtime` without any loss of efficiency.

Reply:
> however, JSOO is clever enough to inline the external declaration across modules, which removes the penalty when targeting Javascript.

This would not apply, I guess, if we use jsoo's separate compilation.  @nojb : is that also your understanding?

Reply:
Haven't tested it, but that would be my guess indeed.

Reply:
This is not my understanding. As long as ocamlc is aware that a symbol is an external, it will be reflected in the bytecode and js_of_ocaml will be able to perform optimisations based on primitive names. It doesn't change anything if the external is declared locally or in a separate compilation unit.

Reply:
the following
```ocaml
let a = Jsoo_runtime.Js.pure_js_expr "thisItExported"`
```
with separate compilation, gets translated into 
```js
// Generated by js_of_ocaml
//# buildInfo:effects=false, kind=cmo, use-js-string=true, version=5.3.0+git-5.3.0-18-gdb77ca4f04

//# unitInfo: Provides: B
//# unitInfo: Requires: Jsoo_runtime
(function
  (globalThis){
   "use strict";
   var runtime = globalThis.jsoo_runtime, a = thisItExported, B = [0, a];
   runtime.caml_register_global(1, B, "B");
   return;
  }
  (globalThis));
```

Reply:
externals exported in the mli of `Jv` need to stay `external` so that when referencing `Jv` from another unit, ocamlc (and jsoo) know about these externals.